### PR TITLE
Transaction

### PR DESCRIPTION
**Pull request by SydDaps**

### CODE REVIEW

File: transaction.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require_relative "customer"
+require_relative "Teller"
```
Comment (s3lase) [inline transaction.rb]: this should be lowercase t

Comment (SydDaps) [reply transaction.rb]: done

File: transaction.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require_relative "customer"
+require_relative "teller"
+
+class Transaction
+  attr_accessor :transaction_type, :customer, :teller, :transaction_amount, :account_type
```
Comment (s3lase) [inline transaction.rb]: This being the in the `Transaction` class, just `type` and `amount` should do as attributes. I think prepending both with `transaction` is redundant.

Also, I'm not too sure about having an `account_type` on the Transaction. what's the rationale behind it?
why not the account object itself or the account number? You're already doing that for customer and teller

Comment (SydDaps) [reply transaction.rb]: the customer may have savings and a maybe a current account so I am taking the account type to know which particular account of the customer  the transaction is being performed on

File: transaction.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  def initialize(fields)
+    @transaction_type = fields[:type]
+    @customer = fields[:customer]
+    @teller = fields[:teller]
+    @transaction_amount = fields[:amount]
+    @account_type = fields[:account_type]
+  end
+
+  def do_transaction
```
Comment (s3lase) [inline transaction.rb]: With a method name like`do_transaction`, a typical usage reads as follows
```ruby
transaction = Transaction.new(...)
transaction.do_transaction
```
`transaction.do_transaction` reads weird. 

A method names like `do`, `perform`, `execute` would read much better

This method also violates the  **Single Responsibility Principle** transaction message generation shouldn't be happening in this method.
What you can do is set and return some form of status/flag and have a separate method for the transaction message

File: transaction.rb (hunk 2 above)
Comment (SydDaps) [reply transaction.rb]: > the customer may have savings and a maybe a current account so I am taking the account type to know which particular account of the customer the transaction is being performed on and also its shown in my return message the type of account that the transaction was done on or I should limit my model to the customer having only one account

File: transaction.rb (hunk 3 above)
Comment (SydDaps) [reply transaction.rb]: oww okay yeah I will do all the changes what of the issue with the account type in there?

**Pull request merged**